### DISCUSSION TITLE
Releasing a new version

### DISCUSSION THREAD

**Discussion by MarijnS95** [ash-rs/ash#376] [General]
Hi @MaikKlein!

It has been quite some time since the last release, and a ton of new Vulkan APIs and features have made it into this repository like the stabilized Ray Tracing API. In addition there are some stale PRs resolving open issues. Is there anything we can help with to get these in and get a new version released?

**Comment by MaikKlein** [comment]
I'll go over the PRs today and release a new version today or tomorrow. Sorry I was a bit time starved.

**Reply by MarijnS95** [reply]
@MaikKlein Don't worry, I don't want to rush you into a release today or tomorrow, but knowing if there's any hold-up and planning towards an eventual release with outstanding issues resolved would be nice :+1:

**Comment by MaikKlein** [comment]
New version is out :partying_face:

**Reply by MarijnS95** [reply]
:partying_face: Thank you so much!

**Comment by MarijnS95** [comment]
@MaikKlein Seeing the new 0.32.1 release, would it be possible to prune the branches a bit? I don't think we need a branch per release, that's what the tags are for. Perhaps a `0.32-stable` track that can be used for a future `.2` if any important fixes need to be backported?

Also, most if not all the commits from master could have probably landed semver-wise: they are only additions and not introducing any breaking changes. Including the bugfix in b3a010a3154d0cd9d4943e794a0e90916da93936.

Finally it looks like the `0.32.1` is one commit behind where it is supposed to be.

**Reply by MaikKlein** [reply]
> Also, most if not all the commits from master could have probably landed semver-wise: they are only additions and not introducing any breaking changes. Including the bugfix in b3a010a.

There is at least one breaking change in master https://github.com/MaikKlein/ash/commit/a2a26fd1e750ad647f07eac8dda36c0168c45828 so I just decided to cherry pick a specific PR and do a patch release for it. 

Not ideal but I also don't want to go to crazy with the backports. Next release will be `0.33` probably.

> Perhaps a 0.32-stable track that can be used for a future .2 if any important fixes need to be backported?

What do you mean with track? A new branch for every "major" release?

**Reply by MarijnS95** [reply]
> There is at least one breaking change in master [a2a26fd](https://github.com/MaikKlein/ash/commit/a2a26fd1e750ad647f07eac8dda36c0168c45828) so I just decided to cherry pick a specific PR and do a patch release for it.

Right! That commit shouldn't land before `0.33`, no.

> Not ideal but I also don't want to go to crazy with the backports. Next release will be `0.33` probably.

Probably. The bug fixed in b3a010a3154d0cd9d4943e794a0e90916da93936 shouldn't hurt too many projects as `with_library` is iirc new since `0.32` (and a pretty niche feature anyway), but I hope users don't end up relying on it :)

> > Perhaps a 0.32-stable track that can be used for a future .2 if any important fixes need to be backported?
> 
> What do you mean with track? A new branch for every "major" release?

Exactly. Most projects seem to create a single parallel branch for every major/minor release to which they pick bugfixes only, and do an occasional patch release from there (with just a tag).

Another question: is the date in `Changelog.md` correct? Says `0.32.0 - 2020-07-03`, should be `2021-03-07` no? And I think you can attach the changelog info to a tag/release on GitHub as well :)

**Reply by MarijnS95** [reply]
Regarding the changelog again, `dependabot` has a hard time figuring out what to post, presumably due to lack of `0.32.1` entry:

![image](https://user-images.githubusercontent.com/2325264/113697183-26e8f800-96d3-11eb-894b-60c195bf50b7.png)

Worth noting that it links to `blob/master`, any changelog updates on a `0.32-release` "stable branch" would have to be synchronized with master too.

**Reply by MarijnS95** [reply]
@MaikKlein Thanks for giving me access to this repo! I've gone ahead and done the following:

- Delete merged branches. Feel free to delete branches for closed PRs or those that are just stale: https://github.com/MaikKlein/ash/branches;
- Fixed the `0.32.0` release date, curated the list a bit (not sure if it's better to maintain commit titles; we might also add commit hashes in front like `- <hash>: <curated title>`, some other projects do that and it's neat to navigate);
- Added `0.32.1` to the changelog;
- Replaced the `new-ash-release` and `0.32.1-release` branch with the aforementioned `0.32-release` branch. Any changes for a future `0.32` patch release go there and can eventually be tagged with `0.32.2` and beyond;
- Pushed tags for `0.32.0`, `0.32.1`, and the `ash-window` `0.5.0` and `0.6.0` release;
- Created "releases" for the tags based on the changelog: https://github.com/MaikKlein/ash/releases. Feel free to edit/improve the title/text.

Apologies for pushing the `0.32.1` tag around a bit; in the end I settled on putting it on the commit listed in the crate source: https://docs.rs/crate/ash/0.32.1/source/.cargo_vcs_info.json points to d044ae5dda640feee7e88003b52da2fc3085bd97.

**Comment by psy-fidelious** [comment]
Maybe time for a new one? There have been quite a few goodies submitted since September.

**Reply by MarijnS95** [reply]
I knew someone would ask this right as my Christmas break started and I'd actually have time to make a release! In fact, right while I was preparing the changelog and the last `1.2.203` header update before making a release!
